src/edu/mills/cs180a/RomanNumeral.java

```diff
@@ -14,154 +14,177 @@
+            }
+        }
+        return numericValue;
+    }
+
+
     /**
-     * Returns the Roman Numeral representation of the given number.
+     * Returns the Roman Numeral representation of any given number.
      *
      * @param n the number to convert
      * @return the Roman Numeral representation
-     * @throws IllegalArgumentException if the number cannot be represented
-     * by a Roman Numeral in the given range
+     * @throws IllegalArgumentException if the argument is out of bounds
      */
     @VisibleForTesting
-    protected static String convertToText(int n) 
-    {
-    	//GOAL: convert number into strings(Roman Numerals)
-    	//Refer to note in convertToInteger
-    	
-    	//acquire input, compare it to the values in the map
-    	//return the key
-    	if(n==MIN_VALUE)
-    	{
-    		LETTERS_TO_VALUES.containsValue(n);
-    	}
-    	
-    	
- 	 	return null;
+    protected static String convertFromInt(int n) {
+        // lowerBound should be set to the Roman Numeral that cosely matches the outermost place
+        // value of any given number. Ex n = 56, lowerBound = L; n = 383, lowerBound = C
+        StringBuilder romanNumeral = new StringBuilder();
+
+        if (n > MAX_VALUE || n < MIN_VALUE) {
+            throw new IllegalArgumentException(n + " is out of bounds");
+        }
+
+        // once the value of n reaches 0, the Roman Numeral string is complete
+        while (n > 0) {
+            // sets variable lowerBound to nearest lower Roman Numeral to be appended to end Roman
+            // Numeral
+            String lowerBound = findLowerBound(n);
+
+            appendLowerBound(romanNumeral, n, lowerBound);
+
+            // Remove multiples of lowerBound from n.
+            n = n % LETTERS_TO_VALUES.get(lowerBound);
+        }
+
+        return romanNumeral.toString();
+
+    }
+
+    // The purpose of this helperMethod is to determine which Roman Numeral should be saved into
+    // lowerBound in convertFromInt.
+    private static String findLowerBound(int n) {
+        for (int i = 0; i < NUMERICS.length-1; i++) {
+            // determine the placement of the numeric value between the most appropriate Roman
+            // Numerals
+            if (NUMERICS[i] <= n && n < NUMERICS[i+1]) {
+                return VALUES_TO_LETTERS.get(NUMERICS[i]);
+            }
+        }
+        // If the numeric value of the integer is 1000 or greater, set lowerBound to M.
+        return VALUES_TO_LETTERS.get(NUMERICS[NUMERICS.length - 1]);
     }
-}//end RomanNumeral class
+
+    // The purpose of this helper method is to append the Roman Numeral symbol, saved into
+    // lowerBound to a StringBuilder. The number of times any symbol is appended to the String
+    // Builder depends on the quotient between the number n and the numeric value of the symbol.
+    private static void appendLowerBound(StringBuilder RN, int n, String symbol) {
```

Comment:
This is an improvement, but the name of the parameter should not be in all caps. That makes it look like a constant.